### DISCUSSION TITLE
Extending Improved Stock Count Draft

### DISCUSSION THREAD

**Discussion by RethaLam** [microsoft/Dynamics365Commerce.Solutions#37] [Q&A] [answered]
We turned on the feature for Improved Stock Count.
This runs now on the CRT side the same code as Inbound operations (InventoryInboundOutboundDocument logic).
However the POS views are still different and I need to be able to extend this view.
I need to add the existing inventory (same value one sees in the On-hand list "Physical inventory" column in HQ) for the specific item and warehouse.
Looking under Developer tools, it looks like the name of the view is: InventoryDocumentStockCountingWorkingView
Ideally, I would also like to add the inventory value to the dialog where one enters the count. the name of this seems to be InventoryDocumentQuantityDialog.

If I look in Pos.Api.d I cannot find an entry for the view above. I only see one for InventoryAdjustmentDocumentWorkingView:
"PosApi/Extend/Views/InventoryAdjustmentDocumentWorkingView" 

Under InboundOutbound I see only an extension point for "PosApi/Extend/Views/InventoryDocumentShippingAndReceivingView".

Can somebody tell me if I am looking for the correct view and if there is any extension points to add a custom column to the view at least. Else I will have to log an extensibility request.

![image](https://user-images.githubusercontent.com/123098075/225399875-ac9bb2fb-b54f-4663-8fcd-b9d7b111781c.png)

Thank you

**Comment by rickwyang** [comment, marked as answer]
Hello Retha, InventoryDocumentStockCountingWorkingView is the correct view for Improved Stock Count feature. Today, it doesn't support to customize the data list columns yet, please log an extensibility request for it. Thanks.

**Comment by RethaLam** [comment]
I will thank you
